main.go

```diff
@@ -92,22 +109,71 @@ func printVersion() {
 }
 
 var (
-	errNoPackageFound = errors.New("no package found")
+	errNoPackageFound        = errors.New("no package found")
+	errUnsupportMultiplePkgs = errors.New(
+		`multiple packages are not supported, 
+		please specify the package ID or directory path that only contains one package`,
+	)
+	errMissingPackageName = errors.New("missing package name")
 )
 
 func generateFromRequest(request GenerationRequest) (string, error) {
 	cfg := packages.Config{
 		Mode: packages.NeedName | packages.NeedTypes,
 	}
-	pkgs, err := packages.Load(&cfg, request.Pkg)
+	if request.ModelPkg == "" {
+		request.ModelPkg = request.Pkg
+	}
+	if request.DestPkg == "" {
+		request.DestPkg = request.Pkg
+	}
+	intfPkgID, err := getPkgID(request.Pkg)
 	if err != nil {
 		return "", err
 	}
-	if len(pkgs) == 0 {
-		return "", errNoPackageFound
+	modelPkgID, err := getPkgID(request.ModelPkg)
```

Comment:
> Function generateFromRequest has 5 return statements (exceeds 4 allowed). New

should i break into smaller function ?

Reply:
If it isn't straightforward to do and does not improve readability, that's ok.